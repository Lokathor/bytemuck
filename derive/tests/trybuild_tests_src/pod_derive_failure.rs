use bytemuck::Pod;

// pod should not work on types with generics
#[derive(Pod)]
#[repr(transparent)]
struct Generic<T>(T);


// should fail because it's not transparent or Repr(c)
#[derive(Zeroable, Pod)]
struct Bad(u16);

// Pod should not be derived by types that need padding
#[derive(Zeroable, Pod)]
#[repr(C)]
struct Padding {
    _byte_one: u8,
    _byte_two: u8,
    _int: u32,
    // this should induce 2 bytes of padding
}

fn main() {}
