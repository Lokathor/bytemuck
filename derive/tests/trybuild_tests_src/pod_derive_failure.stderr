error: cannot find derive macro `Zeroable` in this scope
  --> $DIR/pod_derive_failure.rs:14:10
   |
14 | #[derive(Zeroable, Pod)]
   |          ^^^^^^^^

error: cannot find derive macro `Pod` in this scope
  --> $DIR/pod_derive_failure.rs:14:20
   |
14 | #[derive(Zeroable, Pod)]
   |                    ^^^

error: cannot find derive macro `Zeroable` in this scope
  --> $DIR/pod_derive_failure.rs:10:10
   |
10 | #[derive(Zeroable, Pod)]
   |          ^^^^^^^^

error: cannot find derive macro `Pod` in this scope
  --> $DIR/pod_derive_failure.rs:10:20
   |
10 | #[derive(Zeroable, Pod)]
   |                    ^^^

error: cannot find derive macro `Pod` in this scope
 --> $DIR/pod_derive_failure.rs:4:10
  |
4 | #[derive(Pod)]
  |          ^^^

warning: unused import: `bytemuck::Pod`
 --> $DIR/pod_derive_failure.rs:1:5
  |
1 | use bytemuck::Pod;
  |     ^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
